prototype_source/quantization_in_pytorch_2_0_export_tutorial.rst

```diff
@@ -0,0 +1,350 @@
+                                |
+                         Calibrate/Train
+                                |
+    —--------------------------------------------------------
+    |                      convert_pt2e                     |
+    —--------------------------------------------------------
+                                |
+                    Reference Quantized Model
+                                |
+    —--------------------------------------------------------
+    |                        Lowering                       |
+    —--------------------------------------------------------
+                                |
+            Executorch, or Inductor, or <Other Backends>
+
+Note: ``prepare_pt2e_quantizer`` will be updated to ``prepare_pt2e`` soon.
+
+An existing quantizer object defined for QNNPack/XNNPack is in
+`QNNPackQuantizer <https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py>`__.
+Taking QNNPackQuantizer as an example, the overall Quantization 2.0 flow could be:
+
+::
+
+    import torch
+    import torch._dynamo as torchdynamo
+    from torch.ao.quantization._quantize_pt2e import convert_pt2e, prepare_pt2e
+    import torch.ao.quantization._pt2e.quantizer.qnnpack_quantizer as qq
+
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.linear = torch.nn.Linear(5, 10)
+
+        def forward(self, x):
+            return self.linear(x)
+
+    example_inputs = (torch.randn(1, 5),)
+    model = M().eval()
+
+    # Step 1: Trace the model into an FX graph of flattened ATen operators
+    exported_graph_module, guards = torchdynamo.export(
+        model,
+        *copy.deepcopy(example_inputs),
+        aten_graph=True,
+    )
+
+    # Step 2: Insert observers or fake quantize modules
+    quantizer = qq.QNNPackQuantizer()
+    operator_config = qq.get_symmetric_quantization_config(is_per_channel=True)
+    quantizer.set_global(operator_config)
+    prepared_graph_module = prepare_pt2e_quantizer(exported_graph_module, quantizer)
+
+    # Step 3: Quantize the model
+    convered_graph_module = convert_pt2e(prepared_graph_module)
+
+    # Step 4: Lower Reference Quantized Model into the backend
+
+Inside the Quantizer, we will use the ``QuantizationAnnotation API``
+to convey user's intent for what quantization spec to use and how to
+observe certain tensor values in the prepare step. Now, we will have a step-by-step
+tutorial for how to use the ``QuantizationAnnotation API`` with different types of
+``QuantizationSpec``.
+
+1. Annotate common operator patterns
+--------------------------------------------------------
+
+In order to use the quantized pattern/operators, e.g. ``quantized add``,
+backend developers will have intent to quantize (as expressed by
+`QuantizationSpec <https://github.com/pytorch/pytorch/blob/1ca2e993af6fa6934fca35da6970308ce227ddc7/torch/ao/quantization/_pt2e/quantizer/quantizer.py#L38>`__
```

Comment:
We need to explain what is quantization spec first

Reply:
Thanks. I think it's same as above, so add the explanation of ``QuantizationSpec`` there.